services/core/src/utils/utils.ts

```diff
@@ -62,4 +62,36 @@ export function cborDecode(bytecode: number[]): any {
     const cborLength: number = bytecode[bytecode.length - 2] * 0x100 + bytecode[bytecode.length - 1];
     const bytecodeBuffer = Buffer.from(bytecode.slice(bytecode.length - 2 - cborLength, -2));
     return cbor.decodeFirstSync(bytecodeBuffer);
+}
+
+/**
+ * Asserts that the number of keys of the provided object is expectedSize.
+ * If not, logs an appropriate message (if log function provided) and throws an Error.
+ * @param object the object to check
+ * @param objectName the name of the object to use in error messages
+ * @param expectedSize the size that the object should have
+ * @param log optional log function
+ */
+export function assertObjectSize(object: any, expectedSize: number, log?: Function, info?: any) {
```

Comment:
Can we make this `object` and `info` parameters type safe if it is expecting some particular form of an object.
Also, `@param objectName` that is listed above does not exist in the argument list, while `info` argument is missing in the params list above.

Reply:
Regarding the type of `object`, the idea was to make the function generic in the sense that any JavaScript object may be checked. But as proposed, this function will be a private method of `ValidationService` as it will no longer be required by any other part of implementation once its calling is removed from `VerificationService`. The `log` and `info` parameters will then not have to be passed.